tensorflow/python/kernel_tests/conv_ops_test.py

```diff
@@ -327,11 +327,28 @@ def testConv2DKernelSmallerThanStrideValid(self):
                        expected=expected_output)
 
   def testConv2DKernelSmallerThanStrideSame(self):
-    expected_output = [1, 3, 7, 9]
+    #[[1 4 7 0]
+    # [2 5 8 0]
+    # [3 6 9 0]
+    # [0 0 0 0]]
     self._VerifyValues(tensor_in_sizes=[1, 3, 3, 1],
                        filter_in_sizes=[1, 1, 1, 1],
                        strides=[2, 2], padding="SAME",
-                       expected=expected_output)
+                       expected=[1, 3, 7, 9])
+    # [[1  5  9 13]
+    #  [2  6 10 14]
+    #  [3  7 11 15]
+    #  [4  8 12 16]]
+    self._VerifyValues(tensor_in_sizes=[1, 4, 4, 1],
+                       filter_in_sizes=[1, 1, 1, 1],
+                       strides=[2, 2], padding="SAME",
+                       expected=[1, 3, 9, 11])
+
+    # TODO this currently fails.
+    #self._VerifyValues(tensor_in_sizes=[1, 8, 8, 1],
```

Comment:
We should check what other frameworks do for this case.  I could imagine padding by 1 on both sides so:

[ 1 2 3 4 5 6 7 8 ]
[ 1 2 3 4 5 6 7 8 ]   
...

Your patches for the first row would be:
 [0 0    [0 0   [0 0
  0 1]    4 5]    8 0]

So the output size could be 3x3, rather than 2x2.